macros/sql/deduplicate.sql

```diff
@@ -104,7 +104,10 @@ path: {}
     {% set row_alias = kwargs.get('row_alias') %}
     {% set columns = kwargs.get('columns') %}
 
-    {% if row_alias != None or columns != None %}
+    {% if relation.is_cte is defined and not relation.is_cte %}
```

Comment:
This is a simplified alternative to [this](https://github.com/dbt-labs/dbt-utils/blob/a17d66f332a782a7e81a59f8d249237f3c29612c/macros/jinja_helpers/_is_relation.sql#L2) and [this](https://github.com/dbt-labs/dbt-utils/blob/a17d66f332a782a7e81a59f8d249237f3c29612c/macros/jinja_helpers/_is_ephemeral.sql#L2) to determine if a Relation is backed by a real database object whose columns can be fetched via the `star` macro.

i.e., the columns for tables and views can be retrieved via `information_schema.columns` (or an equivalent), but can't for CTE.